pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -230,5 +327,60 @@ pub mod pallet {
 		}
 	}
 
-	impl<T: Config> Pallet<T> {}
+	impl<T: Config> Pallet<T>
+	where
+		<CurveParameterTypeOf<T> as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign,
+	{
+		fn calculate_collateral(
+			low: CurveParameterTypeOf<T>,
+			high: CurveParameterTypeOf<T>,
+			passive_supply: PassiveSupply<CurveParameterTypeOf<T>>,
+			curve: &Curve<CurveParameterTypeOf<T>>,
+		) -> Result<CollateralCurrencyBalanceOf<T>, ArithmeticError> {
+			let normalized_costs = curve.calculate_costs(low, high, passive_supply)?;
+
+			let collateral_denomination = 10u128
+				.checked_pow(T::CollateralCurrency::decimals(T::CollateralAssetId::get()).into())
+				.ok_or(ArithmeticError::Overflow)?;
```

Comment:
isn't that done more efficiently with a bit shift or something?

Reply:
Bit shifting is doubling up the value right? Not sure if that would help. What I can do is creating a `const fn` to precompute values, but I am not sure, if this would help.

Reply:
yeah sorry, there's no 'bit shift' for decimal numbers is there